### ISSUE TITLE
-lpthread in link_flags reverses link order

### ISSUE THREAD

**Issue by Anomander** [grailbio/bazel-toolchain#102]
Upgrading from [76ce37e977a304acf8948eadabb82c516320e286](https://github.com/grailbio/bazel-toolchain/commit/76ce37e977a304acf8948eadabb82c516320e286) to [ef31d2cedcbb1d41dc86bb94981bca823b3408d1](https://github.com/grailbio/bazel-toolchain/commit/ef31d2cedcbb1d41dc86bb94981bca823b3408d1) we noticed failures in our tests:
```
Error fetching symbol 'pthread_getspecific' from libdl: libpthread_shim.so: undefined symbol: pthread_getspecific
```

[This line](https://github.com/grailbio/bazel-toolchain/blob/master/toolchain/cc_toolchain_config.bzl#L171) moves `-lpthread` to the front of link flags, which causes certain symbols (e.g. `pthread_getspecific`) be removed by the linker and become unreadable from `dlsym`.
```python
            # For single-platform builds, we can statically link the bundled
            # libraries.
            link_flags.extend([
                "-L{}lib".format(toolchain_path_prefix),
                "-l:libc++.a",
                "-l:libc++abi.a",
                "-l:libunwind.a",
                # Compiler runtime features.
                "-rtlib=compiler-rt",
                # To support libunwind.
                "-lpthread",
                "-ldl",
            ])
```

Is there a specific reason why pthread needs to be part of link flags and not link_libs? If so, is there a way to expose a configuration knob which would allow us to configure the opposite?

**Comment by siddharthab** [comment]
Thanks for the issue. I think it's just legacy. Can you please send a PR to do the right thing? 😄

**Comment by Anomander** [comment]
https://github.com/grailbio/bazel-toolchain/pull/103

**Comment by siddharthab** [comment]
Closed by #103.

**Comment by siddharthab** [comment]
I reverted the changes proposed in this issue, because going forward, I think this issue is better solved through the user specifying the new `link_libs` attribute in the repository rule. Please keep in mind when you upgrade bazel-toolchain in your workspace. Please see 7580bbe44efaf6e41b1b77efc7f890ad6b46494b.
